src/axom/core/StaticArray.hpp

```diff
@@ -29,6 +29,74 @@ template <typename T, int N>
 class StaticArray : public StackArray<T, N>
 {
 public:
+  /*!
+   * \brief Constructor
+   */
+  AXOM_HOST_DEVICE StaticArray() : StackArray<T, N>(), m_size(0) { }
+
+  /*!
+   * \brief Copy Constructor
+   * \param obj The object to be copied.
+   */
+  AXOM_HOST_DEVICE StaticArray(const StaticArray &obj)
```

Comment:
NOTE: I added the constructors, destructor, and assignment operators after running into weirdness in a CUDA kernel that clipped Polygons with static-sized buffers. The polygons were not reliably copied.